launch_testing/launch_testing/__init__.py

```diff
@@ -12,258 +12,24 @@
-        defaults to 'output'.
-        :param output_filter: an optional function to filter out i.e. ignore output lines for
-        the test.
-        :param side_effect: an optional side effect of a passing test, currently only 'shutdown'
-        is supported.
-        """
-        assert isinstance(action, ExecuteProcess)
-        test_name = 'test_{}_{}'.format(id(action), test_suffix)
-        out, collate_output, match_output, match_patterns = output_test
-        if not output_filter:
-            output_filter = (lambda x: x)
-        assert any(match_patterns)
-
-        def on_process_exit(event, context):
-            nonlocal match_patterns
-            if any(match_patterns):
-                # Finish test instead of failing to prevent process exit
-                # and process output event handlers from racing.
-                return self._finish(test_name)
-
-        launch_description.add_action(
-            RegisterEventHandler(OnProcessExit(
-                target_action=action, on_exit=on_process_exit
-            ))
-        )
-
-        def on_shutdown(event, context):
-            nonlocal match_patterns
-            if any(match_patterns):
-                process_name = action.process_details['name']
-                reason = 'not all {} output matched!'.format(process_name)
-                self._fail(test_name, reason)
-            self._succeed(test_name, side_effect)
-
-        launch_description.add_action(
-            RegisterEventHandler(OnShutdown(on_shutdown=on_shutdown))
-        )
-
-        def on_process_stdout(event):
-            nonlocal out
-            nonlocal match_patterns
-
-            out = collate_output(out, output_filter(event.text))
-            match_patterns = [
-                pattern for pattern in match_patterns
-                if not match_output(out, pattern)
-            ]
-            if not any(match_patterns):
-                return self._succeed(test_name, side_effect)
-            return None
-
-        launch_description.add_action(
-            RegisterEventHandler(OnProcessIO(
-                target_action=action, on_stdout=on_process_stdout
-            ))
-        )
-        self._arm(test_name)
-
-        return action
-
-    def run(
-        self,
-        launch_service,
-        *args,
-        **kwargs
-    ):
-        """
-        Invoke the `run` method of the launch service.
-
-        :returns: If the return value of the parent method is zero but any of
-          the test processes exited with a non-zero return code the return of
-          the first failed test process is returned.
-        """
-        rc = launch_service.run(*args, **kwargs)
-        if rc == 0:
-            default_rc = 1 if 'failed' in self.__tests.values() else 0
-            rc = next((rc for rc in self.__processes_rc.values() if rc != 0), default_rc)
-        return rc
+from . import tools
```

Comment:
Where is `tools`? I get an `ImportError` here:

```
ros2|sloretz@900991552a31:~/ws/ros2$ launch_test
Traceback (most recent call last):
  File "/home/sloretz/ws/ros2/install/launch_testing/bin/launch_test", line 3, in <module>
    from launch_testing.launch_test_main import launch_test_main
  File "/home/sloretz/ws/ros2/build/launch_testing/launch_testing/__init__.py", line 15, in <module>
    from . import tools
ImportError: cannot import name 'tools'
```

Reply:
I got lost at some point. It's back there now, but thanks for the FYI!